Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,426 @@
+
+    if(cruiseEnableCounter == DEBOUNCE_PERIOD){cruiseEnableButton = true;}
+    else if(cruiseEnableCounter == 0){cruiseEnableButton = false;}
+
+    if(cruiseSetCounter == DEBOUNCE_PERIOD){cruiseSet = true;}
+    else if(cruiseSetCounter == 0){cruiseSet = false;}
+
+    // TOGGLE
+    if(ChargeEnable && regenButton != regenPrevious && regenPrevious){regenToggle = !regenToggle;}
+    regenPrevious = regenButton;
+    
+    if(cruiseEnableButton != cruiseEnablePrevious && cruiseEnablePrevious){cruiseEnable = !cruiseEnable;}
+    cruiseEnablePrevious = cruiseEnableButton;
+
+    #endif
+}
+
+/**
+ * @brief Converts integer percentage to float percentage
+ * @param percent integer percentage from 0-100
+*/
+extern const float pedalToPercent[];
+static float percentToFloat(uint8_t percent){
+    if(percent > 100){
+        return 1.0f;
+    }
+    return pedalToPercent[percent];
+}
+
+/**
+ * @brief Follows the FSM to update the velocity of the car
+*/
+void Task_SendTritium(void *p_arg){
+    OS_ERR err;
+    Minion_Error_t minion_err;
+
+#ifndef __TEST_SENDTRITIUM
+    CANPayload_t velocityPayload;
+            velocityPayload.bytes = 32;
+            velocityPayload.data.d = velocitySetpoint;
+
+    CANPayload_t currentPayload;
+            currentPayload.bytes = 32;
+            currentPayload.data.d = currentSetpoint;
+#endif
+
+    while(1){
+        FSM[state].stateHandler();    // do what the current state does
+        readInputs();   // read inputs from the system
+        FSM[state].stateDecider();    // decide what the next state is
+
+        // Sets brakelight
+        Minion_Write_Output(BRAKELIGHT, brakelightState, &minion_err);
+
+        // Drive
+        #ifndef __TEST_SENDTRITIUM
+         if(MOTOR_MSG_COUNTER_THRESHOLD%MotorCounter == 0){ // if smaller numbers, MOTOR_MSG_COUNTER_THRESHOLD-MotorCounter == 0
+            MotorCounter = 0;
+
+           // MotorController_Drive(velocitySetpoint, currentSetpoint);
+            CANbus_Send(VELOCITY, velocityPayload, CAN_NON_BLOCKING);
+            CANbus_Send(CURRENT_VEC, currentPayload, CAN_NON_BLOCKING);
+
+        }else{
+            MotorCounter++;
+        }
+        #endif
+
+        // Dump Info for Testing
+        #ifdef __TEST_SENDTRITIUM
+        printf("-------------------\n\r");
+        printf("State: %d\n\r", state);
+        printf("cruiseEnable: %d\n\r", cruiseEnable);
+        printf("cruiseSet: %d\n\r", cruiseSet);
+        printf("regenToggle: %d\n\r", regenToggle);
+        printf("brakePedalPercent: %d\n\r", brakePedalPercent);
+        printf("accelPedalPercent: %d\n\r", accelPedalPercent);
+        printf("reverseSwitch: %d\n\r", reverseSwitch);
+        printf("forwardSwitch: %d\n\r", forwardSwitch);
+        printf("currentSetpoint: %f\n\r", currentSetpoint);
+        printf("velocitySetpoint: %f\n\r", velocitySetpoint);
+        printf("velocityObserved: %f\n\r", velocityObserved);
+        printf("-------------------\n\r");
+        #endif
```

Comment:
This should be in a helper function ( such as `dumpInfo()` ).

Reply:
Maybe even with a switch case to print out the name of the state depending on the value of `state`? that might be overkill but otherwise we will need to read like "State: 9" and look up the 9th state in the enum while testing.